### ISSUE TITLE
[enhancement]: add image format with better compression (webp, avif, jxl, etc)

### ISSUE THREAD

**Issue by keturn** [invoke-ai/InvokeAI#2704]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What should this feature add?

We generate a lot of images. I would like them to be encoded more efficiently. I don't need the encoding to be lossless.

I often convert the images I've generated with InvokeAI to webp before sharing them, and it's common for them to come out at **10%** of the size of the PNG, or less. That's not 10% _smaller_, that's 10% _total,_ e.g. a 0.77 MB file to a 0.05 MB file.

Feature-wise, JPEG XL would be my first choice for an output format, as it has fewest constraints on size and plenty of options for bit depth and extra channels.

However, given that InvokeAI is a web-centric application, JPEG XL doesn't have the browser support to be viable anytime soon. [webp](https://caniuse.com/webp) or [avif](https://caniuse.com/avif) would be more practical in this regard.

### Easy part: image data

Write [webp from PIL](https://pillow.readthedocs.io/en/stable/handbook/image-file-formats.html#webp) or [avif from pillow-avif-plugin](https://github.com/fdintino/pillow-avif-plugin).

### Messy part: metadata

InvokeAI currently stores image metadata inside a PNG tEXt chunk. It is not immediately obvious how to apply that to other file formats.

IMHO we should switch to using XMP instead, as XMP works more or less the same way regardless of image format.

**Comment by psychedelicious** [comment]
`good first issue` :D

With nodes, our metadata handling will need to be refactored substantially. Insufficiently forward-thinking decisions on metadata handling could force us to retain multiple layers of backwards compatibility. 

There is some good discussion back in #266 which adds context to the current metadata handling.

We already have two different metadata formats right now - the web UI necessarily diverged from the spec in that issue, while the CLI stuck to it. This topic really needs some brainstorming and careful consideration.

**Comment by psychedelicious** [comment]
Canvas requires lossless images to function.

**Comment by keturn** [comment]
wow stalebot two weeks is a pretty short horizon

yes this is still a relevant enhancement request, though there might be some stuff relating to how nodes send images to storage to do first.

**Comment by keturn** [comment]
Canvas could use some `lossless=True` flag,
which could either stick with PNG or use lossless-mode of one of the more modern (JXL, AVIF, WebP) formats.

I assume this requirement comes from the fact that the canvas workflow involves a lot of encode-decode-encode iterations, and lossy encodings tend to get progressively _more lossy_ with each iteration of that kind of thing?

**Comment by psychedelicious** [comment]
Yes, but also - the canvas consists of images layered on top of one another. Compression can lead to artifacts on the edges of images and visible seams.

I'm generally skeptical of lossy formats in general for professional/enthusiast workflows. Maybe if you just want to make some fun images you'd prefer to save bandwidth, disk, memory etc. But most "serious" workflows will require lossless.

Making lossless a toggle is also risky. Imagine you make it a fair way into a processing chain using lossy compression and then decide you want full quality for the final output. If the lossless isn't retained somewhere, you'd need to redo the whole chain. 

(chain = an arbitrary number of arbitrary processing methods, many of which do not exist at this time)

**Comment by Void2258** [comment]
It would be nice to be able to start with WebP images too. A lot of things you can find on the internet to use as inputs for Img2img are WebP, so this would save lots of conversion. It doesn't need to stay in WebP (ie background convert to PNG is fine), just have it be able to take it without conversion from another program needed.

**Comment by JamesClarke7283** [comment]
WebP images are so much better, more modern features and way more optimised. Support would be much appriciated.
